adagrams/game.py

```diff
@@ -1,13 +1,152 @@
+        letter_strings_ten.append(new_letter) #append the letter to the letter_strings_ten
+        all_letter_strings.remove(new_letter) #remove the letter from all_letter_strings
+        
+    return letter_strings_ten
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #make a copy of the letter_bank
+    letter_bank_copy = letter_bank.copy()
+    # loop over the letter in word and changed all word to upper case
+    for letter in word.upper():
+        if letter in letter_bank_copy: #if letter is in letter_bank_copy
+            letter_bank_copy.remove(letter) #the remove the letter in letter_bank_copy
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    # set a total score to 0
+    total_score = 0
+    # loop over the letter in word and changed all word to upper case
+    for letter in word.upper():
+        total_score += SCORE_CHART[letter] #sum up the score for word
+    if len(word) >= 7 and len(word) <= 10: #if the length of the word is >= 7 and <= 10
+        total_score += 8 #then add 8 points more to the total score
+    return total_score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #set a highest score = 0
+    highest_score = 0
+    #set an empty string to store the highest score word
+    highest_score_word = ""
+
+    #loop over the word in the word list
+    for word in word_list:
+        #set a total_score = 0 for each word
+        total_score = 0
+        #loop over the letter in word and set them upper case
+        for letter in word.upper():
+            #sum up the score for word
+            total_score += SCORE_CHART[letter]
+        #if the length of the word is >= 7 and <= 10
+        if len(word) >= 7 and len(word) <= 10:
+            #then add 8 points more to the total score
+            total_score += 8
+        #if total_score more than highest_score
+        if total_score > highest_score:
+            #replacing the highest_score and highest_score_word with the current word and score
+            highest_score = total_score
+            highest_score_word = word
+        #if they are tie
+        if total_score == highest_score:
+            #check if either word or highest_score_word has length over 10, if so the one with length > 10 will be the highest_score_word
+            if len(word) == 10 and len(highest_score_word) < 10:
+                highest_score_word = word
+            #check when both length < 10, will assign the one with less length to the highest_score_word
+            elif len(word) < 10 and len(highest_score_word) < 10 \
+                and len(word) < len(highest_score_word) : 
+                highest_score_word = word
+            #if they both have the same length, keep the one in highest_score_word since it is the first order in the list
+            elif len(word) == len(highest_score_word):
+                continue
+
+
+    return tuple([highest_score_word, highest_score])
```

Comment:
This works! When we return a tuple, however, we don't need to cast our values to a tuple. We can do it by simply returning both values with a comma between them!

```suggestion
    return highest_score_word, highest_score
```